packages/legacy/core/App/hooks/useOnboardingState.ts

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect, useMemo } from 'react'
+import { useNavigationState } from '@react-navigation/native'
+import { State } from '../types/state'
+import { OnboardingTask } from '../types/navigators'
+import { Config } from '../types/config'
+
+export const useOnboardingState = (
+  store: State,
+  config: Config,
+  termsVersion: number,
+  generateOnboardingWorkflowSteps: (store: State, config: Config, termsVersion: number) => OnboardingTask[]
+) => {
+  const [localState, setLocalState] = useState<OnboardingTask[]>([])
+  const currentRoute = useNavigationState((state) => state?.routes[state?.index])
+
+  const activeScreen = useMemo(() => {
+    return localState.find((s) => !s.completed)?.name
+  }, [localState])
+
+  useEffect(() => {
+    if (!store.stateLoaded) {
+      return
+    }
+
+    const screens = generateOnboardingWorkflowSteps(store, config, termsVersion)
+    setLocalState(screens)
+  }, [store, config, termsVersion, generateOnboardingWorkflowSteps])
+
+  return {
+    localState,
+    setLocalState,
```

Comment:
Does this setter need to be exposed? It doesn't seem to be used anywhere outside of the hook. I'm also wondering if pulling a var called `localState` from a hook into a regular component might be confusing. Any problem with us calling it `onboardingState` or `localOnboardingState` or something like that?

Reply:
Fixed.